plugin.rb

```diff
@@ -63,6 +66,29 @@
     end
   end
 
+  add_to_class(:post, :jira_issue) { custom_fields["jira_issue"] }
+
+  add_to_class(:post, :jira_issue=) do |issue|
+    custom_fields["jira_issue"] = issue
+    save_custom_fields
+    topic.jira_issue_status = issue.dig("fields", "status", "name") if is_first_post?
+  end
+
+  add_to_class(:topic, :jira_issue_status=) do |status|
+    if SiteSetting.tagging_enabled && SiteSetting.discourse_jira_issue_tags_enabled
+      tag_name = "status-#{status.downcase.gsub(/\s+/, "-")}"
+      tag = Tag.find_or_create_by(name: tag_name)
+
+      tag_group = TagGroup.find_by(name: "Jira Issue Status")
+      TagGroupMembership.find_or_create_by(tag_group_id: tag_group.id, tag_id: tag.id)
+
+      if self.topic_tags.where(tag_id: tag.id).empty?
```

Comment:
Can also do this:

```suggestion
      if self.topic_tags.exists?(tag_id: tag.id)
```

Reply:
It looks like you mean to add `!` in the prefix along with `exists?` method. I purposely used the `empty?` method to avoid the usage of `!` here.